### ISSUE TITLE
Add an option to detect the last page and then stop appending (e.g. duplicate pages)

### ISSUE THREAD

**Issue by pa-0** [sixcious/infy-scroll#81]
### Problem

I think it's safe to assume that this is a bug and not user error as I am using the `start=0&view=print` regex snippet that you personally drafted up for me as a solution to my dilemma (where the `start=##` portion of the URL was absent from the first page, only appearing on the second page of each forum thread).  

Using the saved settings you provided, regardless of the length of a given thread, if I continue scrolling to the bottom, rather than actually reaching the end of the thread, I find **Infy**-Scroll living up to its name: it makes every webpage an infinite amount of pages!

### URL

www.autohotkey.com/boards/viewtopic.php?f=6&t=45307&start=0&view=print

### Version

0.8

### Browser

Microsoft Edge for Business Version 120.0.2210.144 (Official build) (64-bit)

### OS

Windows 11 Enterprise

### Device

PC

### Adherence

- [x] I have read the [Sticky](https://github.com/sixcious/infy-scroll/issues/50) and my issue is directly related with the app and is not related to making it work with a specific website

**Comment by sixcious** [comment]
Hello again poa00, thank you for opening this issue! 💜 

I wish this was a bug I could fix for you, but I don't think there is much I can do. I touched on this in [my comment](https://github.com/sixcious/infy-scroll/issues/79#issuecomment-1876670881) in your previous issue.

Basically, if the website decides to respond with the same exact last page no matter how high the start value is, then that's what will continue to be appended. Incrementing is a different beast than following a next link because there is no concept of there being a "last page" unless you reach the maximum possible integer (or decide to stop scrolling).

It is possible for me to try and make Infy "smart" and compare the previous page with the next page, and if they match up exactly, make it stop appending. However, that would be totally unacceptable to many users because it wouldn't allow them to use Infy to scrape a site or to continue incrementing in a situation where there may be duplicate error pages returned (think of consecutive missing IDs that generate the same exact Error 404 page).

I suppose I could make it an option to have it behave that way, but there is really no room left in the UI. It's also such a niche use-case. I'll notate this and keep it in mind in the future, but I'm really trying to limit the number of settings I add to Infy as it has grown quite large. (I also currently have my hands full trying to update everything to Manifest V3 this year, unfortunately!)

Sorry, I know this isn't what you were hoping to hear, but I hope this helps explain things. ☹️

**Comment by poa00** [comment]
Hey no worries.  I appreciate your responsiveness, and the extension is awesome as-is so a minor hiccup like this is not an issue at all.  As a workaround I can just increase the number of pixels from the bottom of the page that triggers the subsequent page to be appended, and print when I reach last page.   

Thinking about your response (the part about your UI not having space): This is just a thought, but you could potentially add a small input box as part of the overlay where a user could enter the total number of pages, after which Infy can stop appending.  

I imagine it wouldn't even have to be something that needs to be saved for later use. Moreover, I bet most users would prefer having to enter two digits on a particular page every time they visit that page, as opposed to the alternative of scrolling in search of the last page of the article/thread to copy/save the article from beginning to its proper end. When it comes to coding, I am definitely a beginner, so I have no idea how much work would be involved in implementing something like this.  I just figured that **if** there's a chance adding this functionality would be feasible, it would be worth mentioning this as a way to avoid further crowding your UI. 

Again, a happy camper here who is very much enjoying the perks of this extension, so however you decide to proceed, I'm definitely not complaining.

**Comment by sixcious** [comment]
Hi again, sorry for the long delay!

I've added a new option that will prevent duplicate pages from being appended in Version 8, which I release last week. It can be found in the Scripts and Styles Dialog. I tested it with the example URL you provided and it stops appending pages after Page 18.

It can also be enabled by default via a hidden setting by going to Infy Scroll's Options Screen > Right Click > Inspect > Console and entering: `chrome.storage.local.set({"duplicateStop": true});`.

I hope this helps. Thanks for using Infy and for making this feature request! 💜
